L126-core-add-alts-google-call-credentials.md

```diff
@@ -1,61 +1,74 @@
-alts-credentials configurations within `grpc_google_default_credentials`. This
-enhancement specifically allows a secondary set of call credentials for ALTS to
+This proposal outlines a change to the gRPC Core C-API to support  
+alts-credentials configurations within `grpc_google_default_credentials`. This  
+enhancement specifically allows a secondary set of call credentials for ALTS to  
 be provided alongside the default credentials.
 
 ## Background
 
-The existing `grpc_google_default_credentials_create` function allows the
-configuration of a single set of call credentials. In some scenarios, a client
-would want to communicate with services that support hard bound credentials over
-ALTS. This proposal addresses the possibility to support this use case on a
+The existing `grpc_google_default_credentials_create` function allows the  
+configuration of a single set of call credentials. In some scenarios, a client  
+would want to communicate with services that support hard bound credentials over  
+ALTS. This proposal addresses the possibility to support this use case on a  
 channel initialized with Google's default credentials.
 
 ## Proposal
 
 ### C-Core changes
 
-This proposal modifies the function `grpc_google_default_credentials_create` to
+This proposal modifies the function `grpc_google_default_credentials_create` to  
 add a second set of call credentials.
 
 ```c
+typedef struct {
+  grpc_call_credentials* secondary_credentials,
+  bool create_hard_bound_credentials = false;
+} grpc_google_default_credentials_options;
+
 GRPCAPI grpc_channel_credentials* grpc_google_default_credentials_create(
     grpc_call_credentials* tls_credentials,
-    grpc_call_credentials* alts_credentials);
+    grpc_google_default_credentials_options* options);
 ```
 
 This new function accepts two arguments:
 
-1.  `tls_credentials`: The primary call credentials, consistent with the
+1.  `tls_credentials`: The primary call credentials, consistent with the  
     existing API. This is usually used for TLS connections.
-2.  `alts_credentials`: A secondary set of call credentials to be used
-    specifically for ALTS connections.
+2.  `options`: A structure that allows the caller to configure specific parameters for  
+    the Default Credentials.
 
-After a secure connection is established, the gRPC runtime identifies the
-connection's transport security type, which indicates whether the underlying
-connection is using a protocol like ALTS or TLS. The runtime then selects the
-appropriate call credentials for that connection. If the determined transport
-security type is ALTS, the provided alts_credentials will be used. For all other
-transport types, the primary call_credentials are used, maintaining the default
+The new struct `grpc_google_default_credentials_options` will be able to initially  
+configure the following parameters:
+
+1.  `secondary_credentials`: A secondary set of call credentials to be used  
```

Comment:
Please document the precedence order of these two options.  What happens if secondary_credentials is non-null but create_hard_bound_credentials is true?